sqlmesh/core/macros.py

```diff
@@ -369,10 +370,31 @@ def add_one(evaluator: MacroEvaluator, column: exp.Literal) -> exp.Add:
 
     registry_name = "macros"
 
+    def __init__(self, name: str = "", coerce_args: bool = False) -> None:
+        super().__init__(name)
+        self.coerce_args = coerce_args
+
     def __call__(
-        self, func: t.Callable[..., DECORATOR_RETURN_TYPE]
+        self,
+        func: t.Callable[..., DECORATOR_RETURN_TYPE],
     ) -> t.Callable[..., DECORATOR_RETURN_TYPE]:
-        wrapper = super().__call__(func)
+        @wraps(func)
+        def _typed_func(*args: t.Any, **kwargs: t.Any) -> DECORATOR_RETURN_TYPE:
+            spec = inspect.getfullargspec(func)
+            kwargs = inspect.getcallargs(func, *args, **kwargs)
+            for param, value in kwargs.items():
+                coercible_type = spec.annotations[param]
+                if isinstance(value, coercible_type):
+                    continue
+                elif issubclass(coercible_type, exp.Expression):
+                    kwargs[param] = parse_one(value.sql(), into=coercible_type)
+                elif coercible_type in (int, float, str) and isinstance(value, exp.Literal):
```

Comment:
i think we can have an option which is if the type is a string, but it's not a literal, you can just call value.sql(dialect=self.dialect)

Reply:
Sure that makes sense. Very convenient.

Reply:
done